File: languages/java/src/ContainsDuplicate.java
import java.util.HashSet;
import java.util.Set;

public class ContainsDuplicate {
    public static boolean containsDuplicate(int[] nums) {
        Set<Integer> seen = new HashSet<>();
        for (int num : nums) {
            if (seen.contains(num)){
                return true;


            }
            seen.add(num);
        }
        return false;

    }


    public static void main(String[] args) {
   int[][] testCases = {
            {1, 2, 3, 1},
            {1, 2, 3, 4},
            {1, 1, 1, 3, 3, 4, 3, 2, 4, 2}
    };
    boolean[] expected = {true, false, true};

    for (int i = 0; i < testCases.length; i++) {
        boolean result = containsDuplicate(testCases[i]);
        System.out.println("nums=" + java.util.Arrays.toString(testCases[i]) +
                " -> result=" + result +
                ", expected=" + expected[i]);
    }

}
}

File: languages/python/src/contains_duplicate.py
def contains_duplicate(nums):
    """
    Given an integer array nums, return true if any value appears at least twice in the array, and return false if every element is distinct.

    :param nums: List[int] - The input array of integers.
    :return: bool - True if any value appears at least twice, False otherwise.
    """
    seen = set()
    for num in nums:
        if num in seen:
            return True
        seen.add(num)
    return False


if __name__ == "__main__":
    test_cases = [
        ([1, 2, 3, 1], True),
        ([1, 2, 3, 4], False),
        ([1, 1, 1, 3, 3, 4, 3, 2, 4, 2], True),
    ]

    for nums, expected in test_cases:
        result = contains_duplicate(nums)
        print(f"nums={nums} -> result={result}, expected={expected}")
